src/knx/table_object.cpp

```diff
@@ -267,6 +273,21 @@ void TableObject::initializeProperties(size_t propertiesSize, Property** propert
                     pushInt(obj->tableReference(), data);
                 return 1;
             }),
+        new CallbackProperty<TableObject>(this, PID_MCB_TABLE, false, PDT_GENERIC_08, 1, ReadLv3 | WriteLv0,
+            [](TableObject* obj, uint16_t start, uint8_t count, uint8_t* data) -> uint8_t {
+                if (obj->_state != LS_LOADED)
+                    return 0; // need to check return code for invalid
+                
+                uint32_t segmentSize = obj->_size;
```

Comment:
I'm a bit confused. Didn't it work with saveSize - foo? Or did you misunderstand my comment? I just wanted to replace "savesize() - 4 -2" with something like "savesize() - sizeof(something) - SOMEDEFINE" so that somebody knows what these numbers mean. You wrote it in the comments, but when code changes, comments often don't.

Reply:
Sorry for the confusion. Yes, it did not work before with saveSize() - 4 - 1. I had a closer look at what saveSize() returns and it is InterfaceObject::saveSize() + 5:
https://github.com/thelsing/knx/blob/e57bbf9dbe25505a9f02d83f111d9df7852c18df/src/knx/table_object.cpp#L230-L233

InterfaceObject::saveSize() is the sum of saveSize() of all properties with writeEnable():
https://github.com/thelsing/knx/blob/e57bbf9dbe25505a9f02d83f111d9df7852c18df/src/knx/interface_object.cpp#L168-L181

For the ApplicationObject this is only PID_PROG_VERSION (which is a DataProperty). Including sizeof(_currentElements) this accounts for 7 bytes (2 + 1*5):
https://github.com/thelsing/knx/blob/e57bbf9dbe25505a9f02d83f111d9df7852c18df/src/knx/data_property.cpp#L113-L116

However, I know that I defined the code segment for parameters in the application object as 256 byte (in knxprod):
`<RelativeSegment Id="M-00FA_A-0000-04-0000_RS-04-00000" Name="Parameters" Size="256" LoadStateMachine="4" Offset="0" />`

So far the only way I found to really get the value of 256 for the size of application object data() is to store it after allocTable(). In my earlier tests the partial programming was working, but only because I only programmed 6 bytes of memory parameters out of the maximum possible 256 and this value=6 was lower than the "incorrectly" returned value of 7 that I mentioned above. Now with _size stored it returns the correct values for segment size in the MCB_TABLE during ETS programming.

Sorry for the rather complicated explanation, I hope it is more clear.

Reply:
You should be able to get the correct size. Let me check.

Reply:
Its possible but a little bit if work: You need to add a method to the Memory class for this. See this line: [https://github.com/thelsing/knx/blob/e57bbf9dbe25505a9f02d83f111d9df7852c18df/src/knx/memory.cpp#L115]

Reply:
I'm also fine with leaving it as it is. But in this case please add a comment why the _size field is needed. Otherwise I might remove it accidentally a year later. ;)

Reply:
Thanks for checking. I think I understand what you mean. I could add a method "Memory::findBlockSizeOfBlock()" which tries to find a MemoryBlock* of the address and then return the size if its not a nullptr (in this case the address is not in _usedList and it should return 0 indicating that the block is not found).

This method could then be called within TableObject with _memory->findBlockSizeOfBlock(_data).

I will try this, challenge accepted ;)

Reply:
Hmm, in principle this works. However, the alignment is a problem:
https://github.com/thelsing/knx/blob/e57bbf9dbe25505a9f02d83f111d9df7852c18df/src/knx/memory.cpp#L155

Here the value of size might be changed. I e.g. got 12 as block->size when allocMemory(10) was initially requested. For CRC calculation this should not matter as data() is allocated up to data()+block->size but the MCB_TABLE segment size would be incorrect then.

Reply:
Ok. Then please add a comment explaining why the _size field is there and why we can't use the size of the memory block and were done.

Reply:
Done. Unfortunately I cannot test what needs to be done to RouterObject. Do we simply have to remove the MCB_TABLE DataProperty and updateMcb() there?

@nanosonde can you comment?

Reply:
I will check. Has been a long time since implementation. 😉

Reply:
Hi!
I have analyzed it. 
Simply removing the MCB_TABLE DataProperty and updateMcb() should be enough.

Reply:
Thanks nanosonde! I removed this as suggested.